Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/WidgetProvider.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WidgetProvider.h"
+#include "WidgetImpl.h"
+
+typedef std::function<std::shared_ptr<WidgetImpl>(winrt::WidgetContext, winrt::hstring)> CreateWidgetImplFn;
+struct WidgetImplCreationInfo
+{
+	CreateWidgetImplFn factoryFn;
+};
```

Comment:
consider just using the fn ptr type

Reply:
added a string to use an std::array rather than a map, this structure now has an extra string